web/client/actions/flexiblelayout.js

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { setControlProperty } from './controls';
+import {
+    updateActivePlugins,
+    updatePanelSizes
+} from '../utils/FlexibleLayoutUtils';
+import {
+    activePluginsSelector,
+    panelSizesSelector,
+    flexibleLayoutStructureSelector
+} from '../selectors/flexiblelayout';
+
+export const updateFlexibleLayoutType = setControlProperty.bind(null, 'flexibleLayout', 'type');
+
+export const resizeFlexibleLayoutPanel = (name, data) => {
```

Comment:
we wanted to avoid thunks as much as possible
 - because testing is hard
 - as well as you need to intercept other actions or do complex things, you may need to switch to epic, so is better to create action creators and epics from the beginning

Anyway in this case we are dispatching one action that sets state in controls. I'd like to use a specific reducer for flexibleLayout instead of controls state, to have more control and visibility on the specific actions and state introduced by this component. Of course this is my opinion, not a strong constraint.